### ISSUE TITLE
PHP-function create_function() has been deprecated (phpseclib 1.0)

### ISSUE THREAD

**Issue by alexgit2k** [phpseclib/phpseclib#1225]
The PHP-function create_function() has been deprecated as of PHP 7.2.0 (https://wiki.php.net/rfc/deprecations_php_7_2#create_function), but is still used in phpseclib 1.0:

```
phpseclib/File/X509.php: $callback = create_function('$x', 'return "\x" . bin2hex($x[0]);');

phpseclib/Crypt/Base.php: if ($this->use_inline_crypt !== false && function_exists('create_function')) {
phpseclib/Crypt/Base.php: *    | callback $inline = create_function:                                                          |
phpseclib/Crypt/Base.php: return create_function('$_action, &$self, $_text', $init_crypt . 'if ($_action == "encrypt") { ' . $encrypt . ' } else { ' . $decrypt . ' }');
```

**Comment by terrafrost** [comment]
The X509 instance was addressed three days ago:

https://github.com/phpseclib/phpseclib/commit/703f7840db0631a52eb761ae55371e08e966f6ca

The Base.php instance is not an issue in either 1.0.9. It was addressed over two months ago (1.0.9 was released about a week ago):

https://github.com/phpseclib/phpseclib/commit/98d46db7c51cf8e4f8648b383d28db78dd22d055

In that diff you can see that it's only doing `return create_function` conditionally:

```php
        // Create the $inline function and return its name as string. Ready to run!
        if (version_compare(PHP_VERSION, '5.3.0') >= 0) {
            eval('$func = function ($_action, &$self, $_text) { ' . $init_crypt . 'if ($_action == "encrypt") { ' . $encrypt . ' } else { ' . $decrypt . ' } };');
            return $func;
        }

        return create_function('$_action, &$self, $_text', $init_crypt . 'if ($_action == "encrypt") { ' . $encrypt . ' } else { ' . $decrypt . ' }');
```

This is further supported by the fact that phpseclib is passing it's unit tests on PHP 7.2:

https://travis-ci.org/phpseclib/phpseclib/builds/311133811

And yes, there are unit tests that would are only passing on 7.2 _because_ `create_function` is no longer used.

**Comment by alexgit2k** [comment]
I am sorry, I missed that. Thank you very much for your fast reply.
